src/main/java/com/boomi/flow/services/boomi/mdh/client/XmlMapAdapter.java

```diff
@@ -1,55 +1,63 @@
         if (wrapper == null || wrapper.elements == null || wrapper.elements.isEmpty()) {
-            return map;
-        }
-
-        for (Object object : wrapper.elements) {
-            Element element = (Element) object;
-
-            if (element.hasChildNodes()) {
-                map.put(element.getNodeName(), createChildNodes(element.getChildNodes()));
-            }
+            return null;
         }
 
-        return map;
+        return new MObject(firstNodeName(wrapper), "", getProperties(wrapper));
     }
 
     @Override
-    public XmlMapWrapper marshal(Multimap<String, Multimap<String, Object>> stringMultimapMap) throws Exception {
+    public XmlMapWrapper marshal(MObject stringMultimapMap) throws Exception {
         throw new RuntimeException("Marshalling maps isn't supported yet");
     }
 
-    private static Multimap<String, Object> createChildNodes(NodeList elements) {
-        Multimap<String, Object> childMap = ArrayListMultimap.create();
+    String firstNodeName(XmlMapWrapper wrapper) {
+        for (Object object : wrapper.elements) {
+            Element element = (Element) object;
+            return element.getNodeName();
+        }
 
-        for (int i = 0; i < elements.getLength(); i++) {
-            Node childNode = elements.item(i);
+        return "";
+    }
+
+    private List<Property> getProperties(XmlMapWrapper wrapper) {
+        List<Property> properties = new ArrayList<>();
 
-            if (childNode.hasChildNodes() == false) {
-                continue;
+        if(wrapper.elements.size() > 0) {
+            Object fieldsEntity = wrapper.elements.get(0);
+            if (fieldsEntity instanceof Element) {
+                properties =  MapAdapterCommon.createPropertiesModel(((Element) fieldsEntity).getChildNodes());
             }
+        }
 
-            if (childNode.getFirstChild().getNodeType() != 1) {
-                childMap.put(childNode.getNodeName(), childNode.getFirstChild().getNodeValue());
-            } else {
-                // it is a list
-                childMap.put(childNode.getNodeName(), createChildNodes(childNode.getChildNodes()));
+        if (wrapper.elements.size() == 2 && properties != null) {
+            // specific for FuzzyMatchDetails
+            Object fieldsFuzzyDetails = wrapper.elements.get(1);
+            if (fieldsFuzzyDetails instanceof Element) {
+                List<Property> fuzzyProperties = MapAdapterCommon.createPropertiesModel(((Element) fieldsFuzzyDetails).getChildNodes());
+
+                if (fuzzyProperties != null && fuzzyProperties.size() ==  6) {
+                    MObject fuzzyMatchDetailsObject = new MObject(FuzzyMatchDetailsConstants.FUZZY_MATCH_DETAILS, fuzzyProperties);
+                    properties.add(new Property(FuzzyMatchDetailsConstants.FUZZY_MATCH_DETAILS, fuzzyMatchDetailsObject));
+                } else {
+                    throw new RuntimeException("fuzzyMatchDetails is a reserved name");
```

Comment:
Can you add some more detail? I don't think it would be clear where or why a user would experience this message. Would something like `The model XYZ in your MDH repository cannot have a property called fuzzyMatchDetails, as it is a reserved property name` work?